### DISCUSSION TITLE
Is there VTK file conversion Windows app similar to one for HDF5?

### DISCUSSION THREAD

**Discussion by DanRRRR** [pyvista/pyvista#3223] [Q&A]
Hello All,
I'm interested with VTK/Silo file converter/extractor similar to the ones existing for HDF5 files. How come there is no one easily found on the entire NET? Not surprisingly barely who knows about these formats.

For example there exist very convenient GUI and Command Prompt converters for HDF4 or 5. Just typing in Command Prompt "h5dump.exe -H file.h5" tells you all about the structure of HDF5 file. And "h5dump.exe -d /particles -o out.dat file.h5" will extract variable called "particle" into ASCII text/binary file

**Comment by DanRRRR** [comment]
By the way somebody expressed opinion that it is impossible to make output of VTK data in text format. Is this true that VTK can plot you image of say your face but you can not get back and can not read the RGB color of each pixel of your image and place it into usual text format???

**Reply by banesullivan** [reply]
Not sure what exactly you are trying to get at here, but you can certainly output VTK data in a text format.

> RGB color of each pixel of your image and place it into usual text format???

I'm not sure exactly what you're trying to do... this is unclear to me. However, I'd bet what you want to do is possible.

**Reply by MatthewFlamm** [reply]
I'm not quite sure either, what the specific question is.  vtk, and consequently pyvista, can certainly output data in text format.  It can also read in binary data, so from this perspective you can accomplish data conversion this way, although file sizes will go way up.  Here is an example:

```py
import pyvista as pv
mesh = pv.Rectangle()  # 1 cell, 4 points
mesh["data"] = [0, 1, 2, 3]  # on points
mesh.save('tmp.vtp', binary=False)
```

snippet from file contents
```txt
<?xml version="1.0"?>
<VTKFile type="PolyData" version="0.1" byte_order="LittleEndian" header_type="UInt32" compressor="vtkZLibDataCompressor">
  <PolyData>
    <Piece NumberOfPoints="4" NumberOfVerts="0" NumberOfLines="0" NumberOfStrips="0" NumberOfPolys="1">
      <PointData Scalars="data">
        <DataArray type="Int64" Name="data" format="ascii" RangeMin="0" RangeMax="3">
          0 1 2 3
        </DataArray>
      </PointData>
      <CellData>
      </CellData>
      <Points>
        <DataArray type="Float64" Name="Points" NumberOfComponents="3" format="ascii" RangeMin="0" RangeMax="1.4142135623730951">
          1 0 0 1 1 0
          0 1 0 0 0 0
REST OF FILE OMMITTED
```

**Reply by DanRRRR** [reply]
I was more interested with ready to use tools (like the apps or EXE files) which do not require any knowledge of VTK, Pythons or programming when you just use them and get the file conversion. For HDF4/5 exist few such tools for Windows and Command prompt which conveniently allow to see the content of the HDF5 files and convert files to text/binary with just a built in options/codes. Same like you do for example when  ZIP/UNZIP the files, you write "ZIP a YourFIle.txt" and get Yourfile.ZIP. Same tools exist with the convenient graphics interfaces.

**Comment by banesullivan** [comment]
VisIt might have what you are looking for: https://visit-dav.github.io/visit-website/

**Reply by DanRRRR** [reply]
Surprisingly, and almost unbeliavably, the Visit (which supports 100 or 200 different formats) and ParaView both having the problems with their Save Data options into text file. I tried with both VTK and Silo - same problem. Visit outputs mesh data with no Values, while Paraview outputs all the values of variables and not the cell coordinates. And even worse, after complaining in the ParaView forum about this defect they responded that this conversion of VTK to text format can not be done (!!!). It's like with VTK file you will plot the image of your face but you can not take the pixel data and save it in text format :)))))))

Without text/binary convertors all these 200 types of formats now in danger of oblivion because recently all compilers have fixed the major problem of binary files - their incompatibility between different compilers and different OS. Now Linux and Windows files are 100% compatible for all existing compilers. This was done as simple as with just these 13 characters by introducing "access=STREAM" option when opening the file. And reading/writing binary files is by far fastest than with any other formats reaching 8 GB/s with the PCIe 4.0

**Reply by DanRRRR** [reply]
Paraview fixed the problem with VTK output. It simply was confused with the fact that data could be cell centered and interface (node) centered in the same file. Coordinates are usually interface centered in most hydro and PIC codes while the values like density, temperature etc cell centered. They made one additional option as a possible solution to interpolate interface centered quantities to cell centered or something like that. 
Visit willing to do that too.
